assets/js/base/context/cart-checkout/payment-methods/use-payment-method-registration.js

```diff
@@ -86,8 +89,13 @@ const usePaymentMethodRegistration = (
 				[ paymentMethod.name ]: paymentMethod,
 			};
 		};
-		for ( const paymentMethodName in registeredPaymentMethods ) {
+
+		for ( let i=0; i<PAYMENT_GATEWAY_SORT_ORDER.length; i++  ) {
+			const paymentMethodName = PAYMENT_GATEWAY_SORT_ORDER[i];
```

Comment:
I liked Mike's idea of trying to preserve order by registration client side (via `registerPaymentMethod`) and I experimented a bit but the logic ends up being more complicated than what we have here (for little gain) so I think doing the loop here is sufficient. The only concern I have here though is that effectively whether a payment method shows up or not is if it's name is present in this `PAYMENT_GATEWAY_SORT_ORDER` setting. Are we sure that this setting is reliable enough? From the usage, I think it's fair to assume it is?

Also, there's some formatting issues with these lines:

```suggestion
		for ( let i = 0; i < PAYMENT_GATEWAY_SORT_ORDER.length; i++ ) {
			const paymentMethodName = PAYMENT_GATEWAY_SORT_ORDER[ i ];
```

Reply:
Thanks for the formatting fix, I'm not sure why this isn't working on commit for me. The hook seems to be running after commit, or it's using the wrong config – I see formatting-only diffs after commit! Will figure that out separately.

> The only concern I have here though is that effectively whether a payment method shows up or not is if it's name is present in this PAYMENT_GATEWAY_SORT_ORDER setting. Are we sure that this setting is reliable enough? From the usage, I think it's fair to assume it is?

Good point - yes we are relying on that. I think that's a safe assumption, it's powered by the same code that used in traditional checkout (`WC()->payment_gateways->get_available_payment_gateways()`). If for some reason a method is lost or the value is empty/missing, it will cause a clear UI bug (no/missing payment methods!) so we should catch that. 

We could put in protection for this, but I don't think that's worthwhile.